src/LodeRunner.API/src/Data/Cache/LRAPICache.cs

```diff
@@ -62,62 +62,78 @@ public LRAPICache(IClientStatusService clientStatusService, ILoadTestConfigServi
         /// <param name="results">Results from cache get.</param>
         /// <param name="logger">Logger.</param>
         /// <returns>IActionResult.</returns>
-        public IActionResult HandleCacheResult<TEntity>(IEnumerable<TEntity> results, NgsaLog logger)
+        public async Task<ActionResult> HandleCacheResult<TEntity>(IEnumerable<TEntity> results, NgsaLog logger)
         {
             // log the request
-            logger.LogInformation(nameof(this.HandleCacheResult), CacheDataRequest);
+            await logger.LogInformation(nameof(this.HandleCacheResult), CacheDataRequest);
 
             if (!results.Any())
             {
-                logger.LogInformation(nameof(this.HandleCacheResult), DataNotFound);
+                await logger.LogInformation(nameof(this.HandleCacheResult), DataNotFound);
 
-                return ResultHandler.CreateResult(DataNotFound, HttpStatusCode.NoContent);
+                return new NoContentResult();
             }
 
-            return this.InternalReturnOKResult(results);
+            return await this.InternalReturnOKResult(results);
         }
 
-        /// <inheritdoc/>
-        public IActionResult HandleCacheResult<TEntity>(TEntity results, NgsaLog logger)
+        ///// <summary>
+        ///// Handles cache results.
+        ///// </summary>
+        ///// <typeparam name="TEntity">Entity type.</typeparam>
+        ///// <param name="results">Results from the cache.</param>
+        ///// <param name="logger">Logger.</param>
+        ///// <returns>
+        ///// Action result.
+        ///// </returns>
+        public async Task<ActionResult> HandleCacheResult<TEntity>(TEntity results, NgsaLog logger)
```

Comment:
Use logger local to class.  The same for the other methods so I won't repeat the comment and then have to resolve them all :)

Reply:
Fixed.